src/js/components/locations-configuration/ZoneAndBinLocations.jsx

```diff
@@ -1,26 +1,126 @@
 import React, { Component } from 'react';
 
 import PropTypes from 'prop-types';
+import { confirmAlert } from 'react-confirm-alert';
+import { getTranslate } from 'react-localize-redux';
 import { connect } from 'react-redux';
 import Alert from 'react-s-alert';
 
+import AddZoneModal from 'components/locations-configuration/AddZoneModal';
+import EditZoneModal from 'components/locations-configuration/EditZoneModal';
 import ZoneTable from 'components/locations-configuration/ZoneTable';
-import Translate from 'utils/Translate';
+import apiClient from 'utils/apiClient';
+import Translate, { translateWithDefaultMessage } from 'utils/Translate';
 
 import 'react-confirm-alert/src/react-confirm-alert.css';
 import 'react-table/react-table.css';
 import 'components/locations-configuration/ZoneTable.scss';
 
+
+const INITIAL_STATE = {
+  zoneData: [],
+};
+
 const PAGE_ID = 'zoneAndBinLocations';
 
+const ZONE = 'ZONE';
+
 class ZoneAndBinLocations extends Component {
   constructor(props) {
     super(props);
+    this.setShowAddZoneModal = this.setShowAddZoneModal.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+    this.addLocation = this.addLocation.bind(this);
+    this.setShowEditZoneModal = this.setShowEditZoneModal.bind(this);
+    this.handleZoneEdit = this.handleZoneEdit.bind(this);
+    this.editLocation = this.editLocation.bind(this);
+    this.deleteLocation = this.deleteLocation.bind(this);
     this.state = {
+      ...INITIAL_STATE,
+      showAddZoneModal: false,
+      showEditZoneModal: false,
+      valuesToEdit: {},
```

Comment:
I think you can find some better name for this, e.g. zoneToEdit or something similar